htdocs/fourn/facture/paiement.php

```diff
@@ -136,7 +136,7 @@
 	        if (substr($key, 0, 7) == 'amount_')
 	        {
 	            $cursorfacid = substr($key, 7);
-	            $amounts[$cursorfacid] = price2num(trim(GETPOST($key)));
```

Comment:
The goald of price2num is to be sure the result is not a float with a long decimal error. So we should not convert it into float.
If you do this, does this fix your problem ?
price2num(trim(GETPOST($key))) ? price2num(trim(GETPOST($key))) : 0

Reply:
@eldy 
"trim" is already execute in GETPOST no ?